testdata/enforce-slice-style-nil.go

```diff
@@ -0,0 +1,92 @@
+package fixtures
+
+func somefn() {
+	m0 := make([]string, 10)
+	m1 := make([]string, 0, 10)
+	m2 := make([]string, 0)    // MATCH /use nil slice declaration (e.g. var args []type) instead of make([]type, 0)/
+	m3 := make([]string, 0, 0) // MATCH /use nil slice declaration (e.g. var args []type) instead of make([]type, 0)/
+	m4 := []string{}           // MATCH /use nil slice declaration (e.g. var args []type) instead of []type{}/
+	m5 := []string{"v1", "v2"}
+	m6 := [8]string{}
+	m7 := [...]string{}
+	var m8 []string
```

Comment:
What about these cases:

```
var m9 []string = make([]string, 0)
var m10 = make([]string, 0)
m11 := []string(nil)
var m12 []string = []string{}
var m13 = []string{}
var m14 = []string(nil)
var m15 []string = nil
```

?

Also, how will this rule handle the following:

```
var m []string
m = make([]string, 0)
m = []string{}
```

? Is it meant to be an issue or on the opposite is not an issue?

And this (here it's artificial but instead of len there can be some meaningful function):

```
	if m := []string(nil); len(m) == 0 {
		fmt.Println("Hello, 世界")
	}
```

Reply:
I've added more test cases to cover all these examples.

> ? Is it meant to be an issue or on the opposite is not an issue?

Currently, it is an issue and I think it's okay for it to be an issue.

Ideally, there would be a separate setting for declarations and assignments but this rule currently doesn't make a distinction between the two. Most style guides focus on slice declaration, but assignments are not well-defined so I think it makes sense to enforce different styles on assignment.

> And this (here it's artificial but instead of len there can be some meaningful function):

All of the rules today allow `nil` assignments, so that would be okay. I'm not sure what the preferred style should be in this case, but it seems like a pretty rare scenario (defining an empty slice in a conditional initializer).